### PR TITLE
[CARBONDATA-473] integrate with spark 2 stable datasource api

### PR DESCRIPTION
**Pull request by scwf**
1.  read path: based on spark2 datasource api
2.  load path: give a tool to do data load

this is a initial version for spark2 integration, and we should continue more refactory work on it in future, here we added 6 tools:
1) clean file
2) compaction
3) load data
4) delete segment by id
5) delete segment by date
6) show segment

### CODE REVIEW

**Comment by jackylk** [comment]
Is there an example to run carbon with spark2?
Can these code work?

File: integration/spark-common/pom.xml [hunk 1]
```diff
@@ -84,6 +79,11 @@
       <artifactId>junit</artifactId>
     </dependency>
     <dependency>
+      <groupId>com.databricks</groupId>
+      <artifactId>spark-csv_${scala.binary.version}</artifactId>
+      <version>1.2.0</version>
```
Comment (scwf) [inline integration/spark-common/pom.xml]: we can extract this as a property

**Comment by scwf** [comment]
the result has some issues for the carbon example:
```
+----------+--------+-----------+-----------+-----------+
|shortField|intField|bigintField|doubleField|stringField|
+----------+--------+-----------+-----------+-----------+
|         1|      11|       null|       44.4|      flink|
|         3|      14|       null|       43.4|       hive|
|         2|      10|       null|       43.4|     impala|
|         4|      16|       null|       42.4|     impala|
|         1|      10|       null|       48.4|      spark|
|         5|      17|       null|       43.4|      spark|
|         1|      10|       null|       43.4|      spark|
|         1|      10|       null|       47.4|      spark|
|         1|      10|       null|       43.4|      spark|
|         1|      10|       null|       43.4|      spark|
+----------+--------+-----------+-----------+-----------+
```

**Comment by jackylk** [comment]
Yes, bigint datatype has some issue, please raise a JIRA for it.
And can you add some basic testcase for this PR

**Comment by scwf** [comment]
@jacklk added test case from empty table

**Comment by jackylk** [comment]
There are codestyle failing, please correct them

**Comment by jackylk** [comment]
CI running
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/715/

**Comment by jackylk** [comment]
CI running
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/716/

**Comment by jackylk** [comment]
CI passed
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/717/

**Comment by jackylk** [comment]
Thanks for working for this

**Pull request closed without merging**